shell/platform/android/io/flutter/view/AccessibilityBridge.java

```diff
@@ -125,6 +126,36 @@ void setAccessibilityEnabled(boolean accessibilityEnabled) {
         }
     }
 
+    private boolean shouldSetCollectionInfo(SemanticsObject object) {
+        // TODO(dnfield): make these lambdas when Java 1.8 support lands.
+        Predicate<SemanticsObject> parentMatcher = new Predicate<SemanticsObject>() {
+            @Override
+            public boolean test(SemanticsObject o) {
+                return o == mA11yFocusedObject;
+            }
+        };
+
+        Predicate<SemanticsObject> flagMatcher = new Predicate<SemanticsObject>() {
+            @Override
+            public boolean test(SemanticsObject o) {
+                return o.hasFlag(Flag.HAS_IMPLICIT_SCROLLING);
+            }
+        };
+
+        // TalkBack expects a number of rows and/or columns greater than 0 to announce
+        // in list and out of list.  For an infinite or growing list, you have to
+        // specify something > 0 to get "in list" announcements.
+        // TalkBack will also only track one list at a time, so we only want to set this
+        // for a list that contians the current a11y focused object - otherwise, if there
+        // are two lists or nested lists, we may end up with announcements for only the last
+        // one that is currently availalbe in the semantics tree.  However, we also want
+        // to set it if we're exiting a list to a non-list, so that we can get the "out of list"
+        // announcement when A11y focus moves out of a list and not into another list.
+        return object.scrollChildren > 0
```

Comment:
I don't quite understand this check:

First part makes sense, we can only do this for lists that we know the counts of. Otherwise we're checking that either:

- This node or its parent has a11y focus
- This node or its parent does not have implicit scrolling.

Are you using the presence of implicit_scroll to determine whether something is a list?

Reply:
I think it would be better to use the presence of a scrollOffset, it's not guaranteed that a lost view has that flag set since it is determined by the ScollPhysics

Reply:
Lost view -> ListView

Reply:
My thought was we'd only ever do list announcements for something with that set, so we only have to worry about it in that case.

Basically, I want to test to see if the next element getting focus will be expecting to announce that you're in a list.  If so, we should let that one register.  If not, we should let the current on eregister so we get an "Out of list" announcement.

Reply:
Hah.  This doesn't make sense because I commited the wrong code.  The check on line 134 is meant to be `== object`

Reply:
Back to the scrolling issue, we shouldn't use implicit scrolling to determine whether something is a list though.

Reply:
I'm basing this off the idea that we would only set in/out of list announcements if has implicit scrolling is true.  https://github.com/flutter/engine/pull/6918/files/b921edb18370fb3378a327b11a90e6e6c5b8a1fb#diff-be56efcdd49538955ac6ed238eda9ed2R310  and we only need this check here if we'd be setting or not setting in/lut of list announcements.  Is line 312 in the wrong spot then?